wallet-new/src/Cardano/Wallet/API/Response.hs

```diff
@@ -269,3 +270,36 @@ instance HasDiagnostic JSONValidationError where
 instance ToServantError JSONValidationError where
     declareServantError _ =
         err400
+
+
+newtype UnsupportedMimeTypeError
+    = UnsupportedMimeTypePresent Text
+    deriving (Eq, Show, Generic)
+
+deriveGeneric ''UnsupportedMimeTypeError
+
+instance ToJSON UnsupportedMimeTypeError where
+    toJSON =
+        jsendErrorGenericToJSON
+
+instance FromJSON UnsupportedMimeTypeError where
+    parseJSON =
+        jsendErrorGenericParseJSON
+
+instance Exception UnsupportedMimeTypeError
+
+instance Arbitrary UnsupportedMimeTypeError where
+    arbitrary =
+        pure (UnsupportedMimeTypePresent "Delivered MIME-type is not supported.")
+
+instance Buildable UnsupportedMimeTypeError where
+    build _ =
+        bprint "The API expects the Content-Type's main MIME-type to be 'application/json'"
```

Comment:
I think this should display what get passed as `Text` to the `UnsupportedMimeTypePresent` constructor ? Don't you think?